server.js

```diff
@@ -32,6 +87,74 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+// TopMusic.find finds all the infomation that I've made an model for.
+app.get('/music', async (req, res) => {
+  const allMusic = await TopMusic.find()
+
+  res.json(allMusic)
+})
+
+app.get('/lists', async (req, res) => {
+  const { page } = req.query
+  const startIndex = 20 * +page
+  const list = await TopMusic.find().skip(startIndex).limit(20).exec()
+
+  res.json(list)
+})
+
+app.get('/genre', async (req, res) => {
+  const queryString = req.query.q
+  const queryRegex = new RegExp(queryString, 'i')
+  // /pop/ is a regex and i makes it search not being case sensitive
+  const genre = await TopMusic.find({ 'genre': queryRegex })
+  if (genre) {
+    //if .find is succesful
+    console.log('Found : ' + genre)
+    res.json(genre);
+  } else {
+    console.log('Error ' + err)
+    res.status(404).json({ message: 'Cannot find this genre', err: err })
+  }
+})
+
+app.get('/artists', async (req, res) => {
+  const artist = await Artist.find().populate('record')
+  res.json(artist)
+})
+
+//this get works with MongoDBs _id
+app.get('/artists/:id', async (req, res) => {
+  const artist = await Artist.findById(req.params.id)
+  if (artist) {
+    res.json(artist)
+  } else {
+    res.status(404).json({ error: 'Artist not found' })
+  }
+})
+
+app.get('/records', async (req, res) => {
+  const record = await Record.find().populate('artist')
+  if (record) {
+    res.json(record)
+  } else {
+    res.status(404).json({ errror: 'Record not found' })
+  }
+})
```

Comment:
I'm having a hard time seeing what the functional difference between `/records` and `/artists` is